cli/blkio.go

```diff
@@ -0,0 +1,185 @@
+	}
+
+	return &types.WeightDevice{
+		Path:   pairs[0],
+		Weight: uint16(weight),
+	}, nil
+}
+
+// Set implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) Set(val string) error {
+	v, err := getValidateWeightDevice(val)
+	if err != nil {
+		return err
+	}
+	w.values = append(w.values, v)
+
+	return nil
+}
+
+// String implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) String() string {
+	var str []string
+	for _, v := range w.values {
+		str = append(str, fmt.Sprintf("%s:%d", v.Path, v.Weight))
+	}
+
+	return fmt.Sprintf("%v", str)
+}
+
+// Type implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) Type() string {
+	return "value"
+}
+
+func (w *WeightDevice) value() []*types.WeightDevice {
+	var weightDevice []*types.WeightDevice
+	for _, v := range w.values {
+		weightDevice = append(weightDevice, v)
+	}
+
+	return weightDevice
+}
+
+// ThrottleBpsDevice defines throttle bps device
+type ThrottleBpsDevice struct {
+	values []*types.ThrottleDevice
+}
+
+func getValidThrottleDeviceBps(val string) (*types.ThrottleDevice, error) {
+	pairs := strings.Split(val, ":")
+	if len(pairs) != 2 {
+		return nil, fmt.Errorf("bad format for throttle device: %s. Correct format should be <device-id>:<rate>, rate unit is optional and can be b/B, k/K, m/M... ", val)
+	}
+
+	rate, err := units.RAMInBytes(pairs[1])
+	if err != nil || rate < 0 {
+		return nil, fmt.Errorf("invalid rate for device %s, rate mustn't be negative", pairs[0])
+	}
+
+	return &types.ThrottleDevice{
+		Path: pairs[0],
+		Rate: uint64(rate),
+	}, nil
+}
+
+// Set implement ThrottleBpsDevice as pflag.Value interface
+func (t *ThrottleBpsDevice) Set(val string) error {
+	v, err := getValidThrottleDeviceBps(val)
+	if err != nil {
+		return err
+	}
+	t.values = append(t.values, v)
+
+	return nil
+}
+
+// String implement ThrottleBpsDevice as pflag.Value interface
+func (t *ThrottleBpsDevice) String() string {
+	var str []string
+	for _, v := range t.values {
+		str = append(str, fmt.Sprintf("%s:%d", v.Path, v.Rate))
+	}
+
+	return fmt.Sprintf("%v", str)
+}
+
+// Type implement ThrottleBpsDevice as pflag.Value interface
+func (t *ThrottleBpsDevice) Type() string {
+	return "value"
+}
+
+func (t *ThrottleBpsDevice) value() []*types.ThrottleDevice {
+	var throttleDevice []*types.ThrottleDevice
+	for _, v := range t.values {
+		throttleDevice = append(throttleDevice, v)
+	}
+
+	return throttleDevice
+}
+
+// ThrottleIOpsDevice defines throttle iops device
+type ThrottleIOpsDevice struct {
+	values []*types.ThrottleDevice
+}
+
+func getValidThrottleDeviceIOps(val string) (*types.ThrottleDevice, error) {
+	pairs := strings.Split(val, ":")
+	if len(pairs) != 2 {
+		return nil, fmt.Errorf("bad format for throttle device: %s. Corrent format should be <device-id>:<rate>, rate unit is optional and can be b/B, k/K, m/M... ", val)
+	}
+
+	rate, err := strconv.ParseUint(pairs[1], 10, 64)
+	if err != nil || rate < 0 {
```

Comment:
I do not think `rate<0` is necessary, because we already use `ParseUint`. And `ParseUint ` returns a value of type `uint64`. You see it cannot be negative. @Ace-Tang